src/metpy/calc/thermo.py

```diff
@@ -359,6 +359,12 @@ def _lcl_iter(p, p0, w, t):
     fp = so.fixed_point(_lcl_iter, pressure.m, args=(pressure.m, w, temperature),
                         xtol=eps, maxiter=max_iters)
     lcl_p = fp * pressure.units
+
+    # Conditional needed due to precision error with np.log in dewpoint.
+    # Causes issues with parcel_profile_with_lcl if removed. Issue #1187
+    if np.isclose(lcl_p, pressure):
+        lcl_p = pressure
```

Comment:
As I look at this change now, I don't understand it.  Doesn't this return the entire pressure profile?

thanks, confused in Iowa.

Reply:
`lcl` only takes scalar input variables (not arrays), so this will only return the pressure provided. The function will error out earlier than this line if an array is provided.

Reply:
Oh thank you!  I was confused as stated since lcl() in 0.11.0 takes arrays.  Whew.

Reply:
it appears to me that `np.isclose` is now where this errors out when passed an array.  FWIW
```
  File "/home/akrherz/projects/MetPy/src/metpy/calc/thermo.py", line 365, in lcl
    if np.isclose(lcl_p, pressure):
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```

Reply:
Well you've just won yourself a bug for the day. I'll open an issue, since if this conditional were not present, it would indeed return an array. We should be catching if arrays are fed in and either take the initial value or return an error. Where did you see it saying that `lcl` takes arrays in 0.11, because I don't see that in the release notes?

Reply:
I didn't see it, I was just naively/ignorantly doing it.  Just taking the same profile object and sequentially passing it to various sounding functions without thinking about it.

Reply:
Ok thanks. I opened up #1199 to discuss this. If you have an opinion about how to handle this case, feel free to comment there.